src/java.base/share/classes/java/lang/String.java

```diff
@@ -3222,6 +3222,109 @@ public String replace(CharSequence target, CharSequence replacement) {
      * @since 1.4
      */
     public String[] split(String regex, int limit) {
+        return split(regex, limit, false);
+    }
+
+    /**
+     * Splits this string around matches of the given regular expression and
+     * returns both the strings and the matching delimiters.
+     *
+     * <p> The array returned by this method contains each substring of this
+     * string that is terminated by another substring that matches the given
+     * expression or is terminated by the end of the string.
+     * Each substring is immediately followed by the subsequence (the delimiter)
+     * that matches the given expression, <em>except</em> for the last
+     * substring, which is not followed by anything.
+     * The substrings in the array and the delimiters are in the order in which
+     * they occur in the input.
+     * If the expression does not match any part of the input then the resulting
+     * array has just one element, namely this string.
+     *
+     * <p> When there is a positive-width match at the beginning of this
+     * string then an empty leading substring is included at the beginning
+     * of the resulting array. A zero-width match at the beginning however
+     * never produces such empty leading substring nor the empty delimiter.
+     *
+     * <p> The {@code limit} parameter controls the number of times the
+     * pattern is applied and therefore affects the length of the resulting
+     * array.
+     * <ul>
+     *    <li> If the <i>limit</i> is positive then the pattern will be applied
+     *    at most <i>limit</i>&nbsp;-&nbsp;1 times, the array's length will be
+     *    no greater than 2 &centerdot; <i>limit</i> - 1, and the array's last
```

Comment:
Hello Raffaello, when this is rendered, it looks like:

<img width="464" alt="centerdot" src="https://user-images.githubusercontent.com/143523/236816797-b8e0f64a-74a7-4d90-bf06-0350a023f24b.png">

I had to read up what `centerdot` is and from what I understand, it's used in mathematical notation for multiplication sign https://en.wikipedia.org/wiki/Interpunct#In_mathematics_and_science. 

However, I don't remember seeing similar usage in our API docs for multiplication symbol. I think, `*` has been widely(?) used in our API docs. Should we use `*` here too?

Reply:
The HTML named entity `&centerdot;` is rendered correctly by the javadoc tool. However, IntelliJ IDEA currently does not recognize it (https://youtrack.jetbrains.com/issue/IDEA-316843). I could replace it by the equivalent `&#xB7;`, which is recognized by IDEA, but which is much less readable in source code.
So, unless there are other reason to prefer `*`, I'd like to stick with `&centerdot;`.

WDYT?

Reply:
Alternatively, one could remove the multiplication sign altogether...

Reply:
> I could replace it by the equivalent `&#xB7;`

I just locally used this as a replacement to `centerdot` and regenerated the javadocs. The generated javadoc HTML looks the same like the screenshot in my previous comment.

My only concern here is that this notation isn't used previously in any of our APIs and since it's part of a "regular" (non mathemical) sentence, it might be hard to understand what it is. I think it is relatable when it's part of some larger mathematical formula/explanation.

I just did a quick search of `centerdot` in the JDK source to see if it's been used in any API documentation currently and didn't find any such cases. So I still think using `*` would be clearer.

Having said that, if there are no objections from Roger or any other reviewers, it's OK with me too.

Reply:
> Alternatively, one could remove the multiplication sign altogether...

Do you mean rewording it to something like:

> the array's length will be no greater than 2 times <i>limit</i> - 1

Reply:
I would replace center-dot with either with "times" or "*" as being easier to read and more familiar in the javadoc.

Reply:
In java.lang.Math I see both "*" and "&times;" used to indicate multiplication.

Reply:
Just replaced with `&times;`, as used in the javadoc of `Double.toString(double)`